cg/clients/chanjo2/api.py

```diff
@@ -0,0 +1,40 @@
+"""Coverage analysis API for human clinical sequencing data."""
+
+import logging
+from typing import Any
+
+import requests
+
+from cg.clients.chanjo2.models import CoverageData, CoverageRequest
+
+LOG = logging.getLogger(__name__)
+
+
+class Chanjo2APIClient:
+    """
+    Chanjo2 API to communicate with a d4tools software in order to return coverage
+    over genomic intervals.
+    """
+
+    def __init__(self, config: dict[str, str]):
+        self.headers = {
+            "Content-Type": "application/json",
+            "accept": "application/json",
+        }
+        self.host = config["chanjo2"]["host"]
+
+    def get_coverage(self, coverage_request: CoverageRequest) -> CoverageData | None:
+        """Send a POST request to the coverage endpoint to retrieve gene coverage summary data."""
+        endpoint: str = f"{self.host}/coverage/d4/genes/summary"
+        post_data: dict[str, Any] = coverage_request.model_dump()
+        try:
+            response = requests.post(url=endpoint, headers=self.headers, json=post_data)
+            response.raise_for_status()
+            response_content: dict[str, Any] = response.json().values()
```

Comment:
It is better to directly parse the response json into a Pydantic model like this [example](https://github.com/Clinical-Genomics/trailblazer/blob/648eeca3683407fd8e1ff75c29f256f0c712d01d/trailblazer/clients/tower/tower_client.py#L28)

Reply:
Assuming that the API actually provides valid JSON data which it does not seem to do?